fl4health/utils/partitioners.py

```diff
@@ -118,19 +145,24 @@ def partition_dataset(self, original_dataset: D, max_retries: int = 5) -> List[D
 
         Returns:
             List[D]: The partitioned datasets, length should correspond to self.number_of_partitions
+
         """
         targets = original_dataset.targets
         assert targets is not None, "A label-based partitioner requires targets but this dataset has no targets"
         partitioned_indices = [torch.Tensor([]).int() for _ in range(self.number_of_partitions)]
 
         partition_attempts = 0
-
+        partitioned_probabilities: Dict = {}
```

Comment:
You can describe this more specifically for mypy as `Dict[T, np.ndarray]` I think?

Reply:
So whenever I do that I get following weird error from mypy ```
Invalid index type "T" for
"dict[T, ndarray[Any, Any]]"; expected type "T"
```
I assume it should be because of generic T type that causes this error.